File: benchmarks/bench_objects.js
// bench_objects.js — Object allocation benchmark
// Measures: heap allocation, GC pressure, field access

class Point {
    constructor(x, y) {
        this.x = x;
        this.y = y;
    }
}

function addPoints(a, b) {
    return new Point(a.x + b.x, a.y + b.y);
}

let p = new Point(0, 0);
for (let i = 0; i < 1000000; i++) {
    const q = new Point(i, i);
    p = addPoints(p, q);
}
console.log(p.x);
console.log(p.y);

File: benchmarks/bench_objects.go
// bench_objects.go — Object allocation benchmark
// Measures: heap allocation, GC pressure, field access
package main

import "fmt"

type Point struct {
	x int
	y int
}

func addPoints(a, b Point) Point {
	return Point{a.x + b.x, a.y + b.y}
}

func main() {
	p := Point{0, 0}
	for i := 0; i < 1000000; i++ {
		q := Point{i, i}
		p = addPoints(p, q)
	}
	fmt.Println(p.x)
	fmt.Println(p.y)
}
